src/useStorageValue/useStorageValue.ts

```diff
@@ -135,6 +145,14 @@ export function useStorageValue<
   options?: UseStorageValueOptions<Type, Initialize>
 ): UseStorageValueResult<Type, Default, Initialize> {
   const optionsRef = useSyncedRef({ ...DEFAULT_OPTIONS, ...options });
+  const parse = (str: string | null, fallback: Type | null): Type | null => {
+    const parseFunction = optionsRef.current.parse ?? defaultParse;
+    return parseFunction(str, fallback);
+  };
+  const stringify = (data: unknown): string | null => {
+    const stringifyFunction = optionsRef.current.stringify ?? defaultStringify;
+    return stringifyFunction(data);
+  };
```

Comment:
The only way to do so that I could see, would be to use `Pick` to explicitly target the options that are used. But that ends up being pointlessly verbose, needing to update the list of keys each time you add a default. I could revisit typing useSyncedRef so DEFAULT_OPTIONS is indirectly checked. I tried that before and had issues, but maybe that was before I switched to these parse/stringify wrappers to deal with other type issues.

Reply:
I think we're good to go like this.